scene/surface-placement/src/main/java/com/esri/samples/surface_placement/SurfacePlacementSample.java

```diff
@@ -168,10 +188,57 @@ public void start(Stage stage) {
       controlsVBox.getChildren()
           .addAll(toggleGroupLabel, drapedBillboardedRadioButton, drapedFlatRadioButton);
 
-      // add the scene view and control box to the stack pane
-      stackPane.getChildren().addAll(sceneView, controlsVBox);
+      // create slider to adjust the z-value
+      Slider zValueSlider = new Slider(0, 150, 70);
+      Label zValueSliderLabel = new Label("Z-Value:");
+      zValueSlider.setShowTickMarks(true);
+      zValueSlider.setShowTickLabels(true);
+      zValueSlider.setMajorTickUnit(50);
+      zValueSlider.setOrientation(Orientation.VERTICAL);
+
+      zValueSlider.valueProperty().addListener((observableValue, oldValue, newValue) -> {
+
+        // get z-value from slider
+        Double zValueFromSlider = zValueSlider.getValue();
```

Comment:
You can get away with using `double`, the primitive type here:

```suggestion
        double zValueFromSlider = zValueSlider.getValue();
```

The IDE does a better job at explaining why than I would ever do:

> The Double class wraps a value of the primitive type double in an object. An object of type Double contains a single field whose type is double.
> In addition, this class provides several methods for converting a double to a String and a String to a double, as well as other constants and methods useful when dealing with a double.